zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -196,15 +233,23 @@ impl TrustedPreallocate for Action {
         // This acts as nActionsOrchard and is therefore subject to the rule.
         // The maximum value is actually smaller due to the block size limit,
         // but we ensure the 2^16 limit with a static assertion.
-        static_assertions::const_assert!(MAX < (1 << 16));
-        MAX
+        // TODO: FIXME: find a way to use it: static_assertions::const_assert!(Self::MAX_ALLOCATION < (1 << 16));
+        AuthorizedAction::<V>::ACTION_MAX_ALLOCATION
     }
 }
 
 impl TrustedPreallocate for Signature<SpendAuth> {
     fn max_allocation() -> u64 {
         // Each signature must have a corresponding action.
-        Action::max_allocation()
+        #[cfg(not(feature = "tx-v6"))]
+        let result = Action::<super::Orchard>::max_allocation();
+
+        // TODO: FIXME: Check this: V6 is used as it provides the max size of the action.
+        // So it's used even for V5 - is this correct?
+        #[cfg(feature = "tx-v6")]
+        let result = Action::<super::OrchardZSA>::max_allocation();
```

Comment:
This is acceptable as long as the max allocation is higher than any protocol limits. `TrustedPreallocate` just prevents denial of service attacks that preallocate a lot of memory by providing a very large list size.